File: strategy/strategy.py
class Duck:
    def __init__(self):
    	self.quackBehavior

    def performQuack(self):
        self.quackBehavior()
    
    def setQuackBehavior(self,f):
        self.quackBehavior = f

def quack():
    print("ガーガー")

def muteQuack():
    print("<沈黙>")

class MallardDuck(Duck):
    def __init__(self):
        self.quackBehavior = quack

mallard = MallardDuck()
mallard.performQuack()
mallard.setQuackBehavior(muteQuack)
mallard.performQuack()

File: strategy/strategy.js
class Duck {
    constructor() {
       let quackBehavior;
    }

    performQuack() {
        this.quackBehavior();
    }
    
    setQuackBehavior(f) {
        this.quackBehavior = f;
    }
};

const quack = () => {
    console.log("ガーガー");
}

const muteQuack = () => {
        console.log("<沈黙>");
}

class MallardDuck extends Duck {
    constructor() {
        super();
        this.quackBehavior = quack;
    }
};

{
    const mallard = new MallardDuck();
    mallard.performQuack();
    mallard.setQuackBehavior(muteQuack);
    mallard.performQuack();
}